### PR TITLE
examples: add FDA training examples (pytorch_lightning, pytorch_lightning_distributed)

### LINKED ISSUE #1956: Add FDA examples
Add training examples for FDA (Fourier Domain Augmentations) based on [Disentangling the Effects of Data Augmentation and Format Transform in Self-Supervised Learning of Image Representations](https://arxiv.org/abs/2312.02205).

The transforms (`FDATransform`, `FDAView1Transform`, `FDAView2Transform`) were added in #1726. What's missing are training examples showing how to use them end-to-end. The model and loss are identical to SimCLR (`NTXentLoss` + `SimCLRProjectionHead`) — only the transform changes — so the examples are straightforward adaptations of the existing SimCLR examples.

### Tasks

- [ ] Add `examples/pytorch/fda.py`
- [ ] Add `examples/pytorch_lightning/fda.py`
- [ ] Add `examples/pytorch_lightning_distributed/fda.py`

Follow `examples/pytorch/simclr.py` as the template, replacing `SimCLRTransform` with `FDATransform`. I'd like to take this on.

### PR DESCRIPTION
**Pull request by saud5150**
Closes #1956

## Description
- [ ] My change is breaking

Adds `examples/pytorch_lightning/fda.py` and
`examples/pytorch_lightning_distributed/fda.py` following their respective
`simclr.py` templates. The model and loss are identical to SimCLR (`NTXentLoss` +
`SimCLRProjectionHead`); only the transform differs. `FDATransform` does not
accept `input_size` directly, so view transforms are passed explicitly — same
pattern as `byol.py` and `tico.py`.

## Tests
- [x] My change is covered by existing tests.
- [ ] My change needs new tests.
- [ ] I have added/adapted the tests accordingly.
- [x] I have manually tested the change.

```
make format-check  # All checks passed
make lint          # All checks passed
```
Ran both examples on CIFAR-10; loss decreased across epochs.

## Documentation
- [ ] I have added docstrings to all public functions/methods.
- [ ] My change requires a change to the documentation (`.rst` files).
- [ ] I have updated the documentation accordingly.
- [x] The autodocs update the documentation accordingly.

### CODE REVIEW

**Review by gabrielfruet — APPROVED** [review]
LGTM! Thank you for your nice contribution!!
